### DISCUSSION TITLE
Question about Blazegraph and RDF4J

### DISCUSSION THREAD

**Discussion by ali3assi** [eclipse-rdf4j/rdf4j#3661] [Q&A]
### Current Behavior

I try to follow the tutorial:
`https://rdf4j.org/documentation/tutorials/maven-eclipse-project/`
and I want to use the `blazegraph` as repository. I also add the dependency. My pom.xml is:

`
```
`<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">`
  <modelVersion>4.0.0</modelVersion>
  <groupId>org.ali</groupId>
  <artifactId>rdf4j_start</artifactId>
  <version>0.0.1-SNAPSHOT</version>
  <name>HelloWorld</name>

<properties>
  <maven.compiler.source>11</maven.compiler.source>
  <maven.compiler.target>11</maven.compiler.target>
</properties>

<repositories>
        <repository>
            <id>bigdata.releases</id>
            <url>http://www.systap.com/maven/releases</url>
        </repository>
    </repositories>


<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>org.eclipse.rdf4j</groupId>
      <artifactId>rdf4j-bom</artifactId>
      <version>3.6.0</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>

<dependencies>
  <dependency>
    <groupId>org.eclipse.rdf4j</groupId>
    <artifactId>rdf4j-storage</artifactId>
    <type>pom</type>
  </dependency>

<dependency>
  <groupId>org.slf4j</groupId>
  <artifactId>slf4j-simple</artifactId>
  <scope>runtime</scope>
</dependency>

        <dependency>
            <groupId>com.bigdata</groupId>
            <artifactId>bigdata</artifactId>
            <version>1.5.1</version>
        </dependency>


</dependencies>
</project>`
```

The pom.xml mentions an error:
![image](https://user-images.githubusercontent.com/16214827/152199824-14e182d4-89af-4357-aace-f9d9bc9253c2.png)

### Expected Behavior

I want to be able to run my code of the tutorial above and connect to the database

### Steps To Reproduce

I am using windows 10 with JDK 14.

### Version

3.6.0

**Comment by barthanssens** [comment]
Looks like the POM is actually OK, but there is an error when downloading the library/artifact from the systap repository.
This is the error I get in Netbeans

```
Could not resolve dependencies for project org.ali:rdf4j_start:jar:0.0.1-SNAPSHOT: Failed to collect dependencies at com.bigdata:bigdata:jar:1.5.1: 
Failed to read artifact descriptor for com.bigdata:bigdata:jar:1.5.1: 
Could not transfer artifact com.bigdata:bigdata:pom:1.5.1 from/to maven-default-http-blocker (http://0.0.0.0/): 
Blocked mirror for repositories: [bigdata.releases (http://www.systap.com/maven/releases, default, releases+snapshots)] 
```

**Reply by barthanssens** [reply]
This is due to the use of plain HTTP when referring to the systap repo, this should work (note the use of HTTPS instead of HTTP)
```
<repositories>
        <repository>
            <id>bigdata.releases</id>
            <url>https://www.systap.com/maven/releases</url>
        </repository>
    </repositories>
````

**Reply by ali3assi** [reply]
Thanks for your reply. Unfortunately, in eclipse still get the same error after changing HTTP by HTTPS
![image](https://user-images.githubusercontent.com/16214827/152211185-76bfdcc9-c407-449b-a383-8f6155918d68.png)

**Comment by barthanssens** [comment]
Hmz ok, are you sure the systap repository is still up and running ?
Because it looks like the Bigdata/Blazegraph has moved to maven central, so if you don't need 1.5 but can use 2.0, you might try removing the "repositories" part altogether, and include bigdata/blazegraph v2.0

See https://github.com/blazegraph/database for the exact name of the dependency

(Als note that the most recent RDF4J version is 3.7.4 instead of 3.6, which is independent of the issue you've mentioned, but 3.7 has some nice improvements so you might want to use that version instead)

**Reply by ali3assi** [reply]
Thanks from rdf4j is working now. Still the part of blazegraph. I dont know if I can find some tutorial on how to integrate rdf4j with blazegraph. I am looking for such solution since I have a huge dataset to hold

Thanks a lot

**Reply by hmottestad** [reply]
Blazegraph has some very very old samples here: https://github.com/blazegraph/blazegraph-samples

It was my understanding that Amazon bought blazegraph and then ditched the open source project. I think they called it Amazon Neptune.

Instead of Blazegraph I would recommend you try out GraphDB if you want something more robust and scalable than the NativeStore in RDF4J. https://graphdb.ontotext.com/documentation/enterprise/using-graphdb-with-the-rdf4j-api.html

**Reply by ali3assi** [reply]
Thansk for this infos. I wanna try GraphDB
